Allow copying of text from media messages.

greyson-signal opened the pull request:
Previously, the copy option was disabled for all media messages, even if they have text. This change enables the copy option for media messages that have text. Only the text is copied, not the media itself.

Fixes #7472.

**Test Cases**
* Copy regular message with text
* Copy option appears and copies text on media message with text
* Copy option does not appear on media message without text
* Sent media message with whitespace as the text -- whitespace is trimmed, no copy option appears after sending
* Copy option appears when multiple messages are selected, including media messages
* Copy multiple messages at once, including media messages both with and without text
  * Message with no text first, in the middle, and last -- verified pasted text had no extra newlines

**Test Devices**
* [OnePlus One, Android 5.0.2](https://www.gsmarena.com/oneplus_one-6327.php)

FeuRenard commented:
It's not possible to copy the text of multiple messages (including at least one with media) in one go after this change.

moxie0 commented on src/org/thoughtcrime/securesms/ConversationFragment.java:
```diff
@@ -230,7 +223,7 @@ private void setCorrectMenuVisibility(Menu menu) {
       menu.findItem(R.id.menu_context_save_attachment).setVisible(false);
       menu.findItem(R.id.menu_context_resend).setVisible(false);
-      menu.findItem(R.id.menu_context_copy).setVisible(!actionMessage  && !mediaMessage);
+      menu.findItem(R.id.menu_context_copy).setVisible(!actionMessage);
```
Doesn't this mean that selecting two media messages with no text will expose the "copy" option?

FeuRenard replied:
That's why I introduce a variable `emptyBody` to hide the copy button as soon as there is a message without text selected in PR #7519.

moxie0 commented on src/org/thoughtcrime/securesms/ConversationFragment.java:
```diff
@@ -242,7 +235,7 @@ private void setCorrectMenuVisibility(Menu menu) {
       menu.findItem(R.id.menu_context_forward).setVisible(!actionMessage);
       menu.findItem(R.id.menu_context_details).setVisible(!actionMessage);
-      menu.findItem(R.id.menu_context_copy).setVisible(!actionMessage && !mediaMessage);
+      menu.findItem(R.id.menu_context_copy).setVisible(!actionMessage && messageRecord.getBody().length() > 0);
```
Seems strange that selecting a media message with no text that's after a message with text will not expose copy, but the inverse will?

greyson-signal replied:
So right now the logic is: if there's only one message selected, only show copy if there's text -- if >1 are selected, always show copy. I can change it so we always display copy based on whether there is text present.

greyson-signal commented:
Okee dokee, updated is so that we check to make sure that _some_ message has text in order to show 'copy'. Also, found there was a little bug with the text that was copied -- it'd put a newline for messages with no text. Fixed that too.

moxie0 commented:
in 4.19.0
